sdk/core/azure-core/inc/context.hpp

```diff
@@ -39,6 +39,13 @@ namespace Azure { namespace Core {
     };
 
   public:
+
+// The compiler gives a false positive on the union type, disable that warning for this constructor
+#if _MSC_VER
+#pragma warning(push)
+#pragma warning(disable:26495)
+#endif
+
```

Comment:
Commenting on pre-existing issues: This header should `#include <new>` for the non-allocating placement `new` invoked on lines 78 and 81.

Reply:
The `unique_ptr` constructor template on lines 57-65 is overcomplicated; given that `unique_ptr<Derived>` is implicitly convertible to `unique_ptr<Base>` (https://godbolt.org/z/ah1EPz) it could simply be:
```c++
ContextValue(std::unique_ptr<ValueBase>&& p) noexcept
        : m_contextValueType(ContextValueType::UniquePtr), m_p(std::move(p))
    {
    }
```

Reply:
@CaseyCarter , the ValueBase  has a virtual destructor which breaks the implicit conversion.  This change made it impossible to add derived classes. Only allowing ValueBase class to be added

Reply:
see: https://github.com/Azure/azure-sdk-for-cpp/pull/1796